modules/cms/classes/Controller.php

```diff
@@ -1605,4 +1626,27 @@ protected function verifyCsrfToken()
             $token
         );
     }
+
+    /**
+     * Removes '@' from soft component name
+     * @param string $label
+     * @return string
+     */
+    protected function parseComponentLabel($label)
+    {
+        if ($this->isSoftComponent($label)) {
+            return str_replace('@', '', $label);
+        }
+        return $label;
+    }
+
+    /**
+     * Checks if component name has @.
+     * @param string $label
+     * @return bool
+     */
+    protected function isSoftComponent($label)
+    {
+        return $label[0] === '@';
```

Comment:
@DanHarrin Please change this to use `substr` instead to make the intention of this line more clear.
```suggestion
        return substr($label[0], 0, 1) === '@';
```